eventmesh-dashboard-core/src/main/java/org/apache/eventmesh/dashboard/core/meta/runtime/NacosRuntimeCore.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.dashboard.core.meta.runtime;
+
+import org.apache.eventmesh.dashboard.common.model.metadata.RuntimeMetadata;
+import org.apache.eventmesh.dashboard.common.model.remoting.runtime.GetRuntimeRequest;
+import org.apache.eventmesh.dashboard.common.model.remoting.runtime.GetRuntimeResponse;
+import org.apache.eventmesh.dashboard.common.model.remoting.runtime.GetRuntimeResult;
+import org.apache.eventmesh.dashboard.core.function.SDK.SDKManager;
+import org.apache.eventmesh.dashboard.core.function.SDK.SDKTypeEnum;
+import org.apache.eventmesh.dashboard.core.function.SDK.config.CreateNacosConfig;
+import org.apache.eventmesh.dashboard.service.remoting.MetaRemotingService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+import com.alibaba.nacos.api.exception.NacosException;
+import com.alibaba.nacos.api.naming.pojo.Instance;
+import com.alibaba.nacos.client.naming.NacosNamingService;
+
+public class NacosRuntimeCore implements MetaRemotingService {
+
+    @Override
+    public GetRuntimeResult getRuntime(GetRuntimeRequest getRuntimeRequest) {
+        CreateNacosConfig createNacosConfig = new CreateNacosConfig();
+        createNacosConfig.setServerAddress(getRuntimeRequest.getRegistryAddress());
+        NacosNamingService nacosNamingService = (NacosNamingService) SDKManager.getInstance()
+            .createClient(SDKTypeEnum.META_NACOS_NAMING, createNacosConfig).getValue();
+        GetRuntimeResult getRuntimeResult = new GetRuntimeResult();
+
+        CompletableFuture<GetRuntimeResponse> future = CompletableFuture.supplyAsync(() -> {
+            try {
+                List<Instance> instances =
+                    nacosNamingService.getAllInstances("EVENTMESH-runtime-GRPC", "GRPC-GROUP");
```

Comment:
> I would suggest to unify protocols with the same IP to one Runtime instance.
what do you mean? when we get runtime instance from meta, we don't know which protocal to choose.

Reply:
> we don't know which protocal to choose.

That's why Dashboard should **try** to get all protocols' server from Meta.

If you are going to operate Runtime instances by **IP** instead of protocol servers, you can treat servers that have multiple protocols enabled as the same Runtime instance.

Reply:
ok

Reply:
resolve in [22bf6d9](https://github.com/apache/eventmesh-dashboard/pull/98/commits/22bf6d9bbfc9d51ce3b5006cacd6917147d37608)